engine/consensus/compliance/core.go

```diff
@@ -365,7 +373,7 @@ func (c *Core) processBlockProposal(proposal *flow.Block) error {
 	if err != nil {
 		if state.IsInvalidExtensionError(err) {
 			// if the block proposes an invalid extension of the protocol state, then the block is invalid
-			return engine.NewInvalidInputErrorf("invalid extension of protocol state (block: %x, height: %d): %w", blockID, header.Height, err)
+			return model.NewInvalidProposalErrorf(hotstuffProposal, "invalid extension of protocol state (block: %x, height: %d): %w", blockID, header.Height, err)
```

Comment:
same comment here - we are potentially discarding the information required to produce a slashing challenge